File: editor-sora/src/main/java/com/raredev/vcspace/editor/completion/CompletionProvider.java
package com.raredev.vcspace.editor.completion;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public abstract class CompletionProvider {

  private static final Map<Class<?>, CompletionProvider> providers = new HashMap<>();

  public abstract List<VCSpaceCompletionItem> getCompletions(CompletionParams params);

  public static void registerCompletionProviders() {
    if (providers.isEmpty()) {
      
    }
  }

  public static CompletionProvider getCompletionProvider(Class<?> clss) {
    return providers.get(clss);
  }
}

File: editor/src/main/java/com/raredev/vcspace/editor/completion/CompletionProvider.kt
package com.raredev.vcspace.editor.completion

abstract class CompletionProvider {
    abstract fun getCompletions(params: CompletionParams?): List<VCSpaceCompletionItem?>?

    companion object {
        private val providers: Map<Class<*>, CompletionProvider> = HashMap()
        fun registerCompletionProviders() {
            if (providers.isEmpty()) {

            }
        }

        fun getCompletionProvider(clss: Class<*>): CompletionProvider? {
            return providers[clss]

        }
    }
}